### DISCUSSION TITLE
HTTP Basic Auth

### DISCUSSION THREAD

**Discussion by devabdulharis** [aldinokemal/go-whatsapp-web-multidevice#149] [General]
gimana cara mengaktifkan http basic auth ya?

**Comment by aldinokemal** [comment]
tinggal tambahin aja `--basic-auth=user:password`

**Reply by devabdulharis** [reply]
oke done bang.
btw mau send image via API gagal terus di postman imagenya string diisi url kok ga bisa ya? apa base64 (binary) maksudnya? kalau via UI bisa.

{
    "code": "INTERNAL_SERVER_ERROR",
    "message": "failed to decode: schema: converter not found for multipart.FileHeader"
}

**Reply by aldinokemal** [reply]
dilihat aja bro codingnya di UI, bisa di contoh

**Reply by blazzid** [reply]
basic-auth berhasil, tetapi kenapa saat jalankan API malah error "unauthorized" ?
tetapi kalau tanpa basic-auth API success. 

curl \
 -X POST http://localhost:3000/send/message \
 -H "Content-Type: application/json" \
 -d '{"phone":"6289685028129@s.whatsapp.net","message":"selamat malam","reply_message_id":"3EB089B9D6ADD58153C561"}'

apa ada penambahan auth di apinya ?

**Reply by aldinokemal** [reply]
```bash
curl 'http://localhost:3000/user/my/groups' \
  -H 'Accept: application/json, text/plain, */*' \
  -H 'Authorization: Basic dXNlcm5hbWU6ZXhhbXBsZVBhc3N3b3Jk'
```

ini bro contohnya, 
u: username
p: examplePassword

<img width="381" alt="image" src="https://github.com/aldinokemal/go-whatsapp-web-multidevice/assets/14232125/16b129f2-85f1-47b5-8607-a617752340f2">


detailnya baca disini aja ya https://stackoverflow.com/a/1386117

**Comment by aanmochtar80** [comment]
apa untuk di docker sama juga --basic-auth=user:password ? karena saya coba tambahkan baris itu malahan eror. mohon petunjuknya bang

**Reply by aldinokemal** [reply]
basic auth itu base64(user:password)
https://stackoverflow.com/a/1386117 @aanmochtar80

**Reply by aanmochtar80** [reply]
> basic auth itu base64(user:password) https://stackoverflow.com/a/1386117 @aanmochtar80

scriptnya jadi bagaimana yah kalau run pakai docker om?

**Reply by aldinokemal** [reply]
`docker run --detach --publish=3000:3000 --name=whatsapp --restart=always --volume=$(docker volume create --name=whatsapp):/app/storages aldinokemal2104/go-whatsapp-web-multidevice --basic-auth=user:password`

**Reply by aanmochtar80** [reply]
> `docker run --detach --publish=3000:3000 --name=whatsapp --restart=always --volume=$(docker volume create --name=whatsapp):/app/storages aldinokemal2104/go-whatsapp-web-multidevice --basic-auth=user:password`

sudah saya coba dan berhasil tapi pada saat menautkan perangkat kok jadi tidak bisa yah? apa ada yang harus di configurasi lagi?

**Comment by unyil46** [comment]
Kalau untuk go, scriptnya bagaimana? Apa di setting.go, tapi masih bingung variabelnya bagaimana

**Reply by unyil46** [reply]
Dah ketemu 
Baris
`AppBasicAuthCredential string`
diganti
`AppBasicAuthCredential ="username:password"`
